lib/datadog/profiling/collectors/info.rb

```diff
@@ -15,7 +15,15 @@ module Collectors
       # gathering and easily support more flexible/dynamic info collection in the future.
       class Info
         def initialize(settings)
-          @profiler_info = nil
+          # Steep: This one is weird, we set it to nil here but it is always set
+          # to type `profiler_info` in the collect_profiler_info method.
+          # We type it as `profiler_info` in the .rbs file, but set it to `profiler_info?` here
+          # to satisfy the type checker.
+          # Because we type it as `profiler_info` in the .rbs file, the type checker will
+          # complain that the branch in the collect_profiler_info method is unreachable.
+          # But it is just :information level, so we can ignore it.
+          @profiler_info = nil # @type ivar @profiler_info: profiler_info?
```

Comment:
Minor: I'm OK with doing it this way although we could change `collect_profiler_info` to use a `||= begin ... ` to match the other methods and allow us to remove this.

(I believe this initialization was only added because the `unless @profiler_info` check would generate a warning in some Ruby versions, but using `||= begin` will not generate the warning so it's a feasible replacement as well).